Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,221 @@
+
+class NexusGeometrySaveTest : public CxxTest::TestSuite {
+
+private:
+  std::pair<std::unique_ptr<Mantid::Geometry::ComponentInfo>,
+            std::unique_ptr<Mantid::Geometry::DetectorInfo>>
+      m_instrument;
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static NexusGeometrySaveTest *createSuite() {
+    return new NexusGeometrySaveTest();
+  }
+  static void destroySuite(NexusGeometrySaveTest *suite) { delete suite; }
+
+  NexusGeometrySaveTest() {
+    auto instrument = ComponentCreationHelper::createMinimalInstrument(
+        Mantid::Kernel::V3D(0, 0, -10), Mantid::Kernel::V3D(0, 0, 0),
+        Mantid::Kernel::V3D(1, 1, 1));
+
+    m_instrument =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+  }
+
+  void test_providing_invalid_path_throws() { // deliberately fails
+
+    std::string path = "invalid_path"; // test invalid path
+
+    TS_ASSERT_THROWS(saveInstrument(*m_instrument.first, path),
+                     std::invalid_argument &);
+  }
+
+  void test_progress_reporting() {
+
+    MockProgressBase progressRep;
+    EXPECT_CALL(progressRep, doReport(testing::_)).Times(1);
+    ScopedFileHandle test("testFile.hdf5");
+    std::string path = test.fullPath();
+    saveInstrument(*m_instrument.first, path, &progressRep);
+    TS_ASSERT(testing::Mock::VerifyAndClearExpectations(&progressRep));
+  }
+
+  void test_extension_validation() {
+
+    ScopedFileHandle fileResource(
+        "testIstrument.abc"); // creates a temp directory for the file.
+    std::string destinationFile = fileResource.fullPath();
+
+    // TODO the following should not be tested. but saveInstrument should be.
+    TS_ASSERT_THROWS(saveInstrument(*m_instrument.first, destinationFile),
+                     std::invalid_argument &);
+  }
+
+  void test_nxinstrument_class_exists() {
+
+    ScopedFileHandle fileResource(
+        "testIstrument.hdf5"); // creates a temp directory for the file.
+    std::string destinationFile = fileResource.fullPath();
+    saveInstrument(*m_instrument.first,
+                   destinationFile); // saves the instrument.
+
+    HDF5FileTestUtility tester(
+        destinationFile); // tests the file has given attributes.
+
+    TS_ASSERT(tester.hasNxClass("NXinstrument", "/raw_data_1/instrument"));
+    TS_ASSERT(tester.hasNxClass("NXentry", "/raw_data_1"));
+  }
+
+  void test_instrument_has_name() {
+
+    ScopedFileHandle scopedFile("instrument.hdf5"); // temp directory
+    auto destinationFile = scopedFile.fullPath();
+
+    saveInstrument(*m_instrument.first, destinationFile); // saves instrument
+    HDF5FileTestUtility testUtility(destinationFile);
+
+    // TS_ASSERT(testUtility.hasDataSet("/raw_data1/instrument/name", NX_CHAR));
+    TS_ASSERT(testUtility.hasDataSet("name", "short_name", "NX_CHAR",
```

Comment:
You are testing the wrong functionality. We do not expect that `short_name` has value "NX_CHAR"

Reply:
fixed